contracts/contracts/FleekPausable.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+error ContractIsPaused();
+error ContractIsNotPaused();
+error ContractIsNotPausable();
+error PausableIsSetTo(bool state);
+
+abstract contract FleekPausable is Initializable {
+    /**
+     * @dev Emitted when the pause is triggered by `account` and set to `isPaused`.
+     */
+    event PausedChange(bool indexed isPaused, address account);
+
+    /**
+     * @dev Emitted when the pausable is triggered by `account` and set to `isPausable`.
+     */
+    event PausableChange(bool indexed isPausable, address account);
+
+    bool private _paused;
+    bool private _canPause; // TODO: how should we verify if the contract is pausable or not?
+
+    /**
+     * @dev Initializes the contract in unpaused state.
+     */
+    function __FleekPausable_init() internal onlyInitializing {
+        _paused = true;
```

Comment:
Doesn't this line contradict the comment on line 27?

Reply:
Yeah, going to fix the comment

Reply:
Wait I think it should start in unpaused state right? Not paused but can pause would be the init state imo.

Reply:
But in that case I think at the start we would just not deploy the contract until release moment.  I can't think of a scenario where we would want to release it but pause it at start so nobody can use it.  I noticed the Zeppelin example starts unpaused too.  Also if we leave it paused we need a second transaction just to start it.  Maybe we can make the constructor take the start pause state so we can be flexible with it?

Reply:
I think we can have it unpaused by default, I don't think we need a flexible start for it tbh. I'm going to change it